impersonate_login/models/model.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2024 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.http import request
+
+
+class BaseModel(models.AbstractModel):
+    _inherit = "base"
+
+    @api.model_create_multi
+    def _create(self, data_list):
+        res = super()._create(data_list)
+        if (
+            request
+            and request.session.impersonate_from_uid
+            and "create_uid" in self._fields
+        ):
+            impersonator_id = request.session.impersonate_from_uid
+            for rec in res:
+                self.env.cr.execute(
+                    """
+                    UPDATE "%s" SET create_uid = %%s WHERE id = %%s
+                """
+                    % self._table,
+                    (impersonator_id, rec.id),
+                )
```

Comment:
**issue**: Prefer `%(varname)` syntax if possible please. https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#33idioms

**question**: Is there a strong reason for running a query directly instead of doing it like in the original commit?

```python
            for rec in res:
                rec["create_uid"] = request.session.impersonate_from_uid
```

If we have to run the query directly, can we at least run a single query for all records in `res` instead of a separate query for each?

Reply:
I should have used the preferred syntax now.

> **question**: Is there a strong reason for running a query directly instead of doing it like in the original commit?

I was not able to get it working in the way was done in the original commit because the assignment of `rec["create_uid"]` doesn't not happen for some reason.

I think has something to do with the method `_create` that in the following version of this module is changed with `_prepare_create_values` but I'm not sure.

Reply:
ah, good to know, thank you so much @amh-mw

Reply:
@amh-mw Thanks for the explanation!